### PR TITLE
Create api to override config url and default to prod

### PR DESCRIPTION
**Pull request by achocron**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Created API to override config url and default to prod and added test app functionality

I changed some "url"s to "URL" as I encountered them, but I created a task for the main work to do that. (bug 58452)

### CODE REVIEW

File: AppCenterIdentity/AppCenterIdentityTests/MSIdentityTests.m [hunk 1]
```diff
@@ -979,10 +981,33 @@ - (void)testSignOutDoesNothingWhenDisabled {
...
+  XCTAssertEqualObjects([[MSAuthTokenContext sharedInstance] authToken], authToken);
+  XCTAssertEqualObjects([MSMockKeychainUtil stringForKey:kMSIdentityAuthTokenKey], authToken);
+  XCTAssertEqualObjects([self.settingsMock objectForKey:kMSIdentityMSALAccountHomeAccountKey], accountId);
   [identityMock stopMocking];
 }
 
+- (void)testDefaultConfigURL {
+
+  // Then
+  XCTAssertEqualObjects(self.sut.configURL, kMSIdentityDefaultBaseURL);
```
Comment (guperrot) [inline AppCenterIdentity/AppCenterIdentityTests/MSIdentityTests.m]: can you make that test the same as the next one but without calling setConfigUrl to test the default?

File: Sasquatch/Sasquatch/Constants.h [hunk 2]
```diff
@@ -36,3 +36,5 @@ static int const kMSStoragePageSize = SQLITE_DEFAULT_PAGE_SIZE;
 #else
 static int const kMSStoragePageSize = 4096;
 #endif
+
+static const NSString *kMSIntConfigUrl = @"https://mobilecentersdkdev.blob.core.windows.net";
```
Comment (guperrot) [inline Sasquatch/Sasquatch/Constants.h]: that's going away, the one you're looking for is `https://config-integration.dev.avalanch.es` you might want to copy the blobs there as part of this task to avoid breaking our test apps.

File: Sasquatch/SasquatchSwift/AppCenterDelegateSwift.swift [hunk 3]
```diff
@@ -49,6 +49,12 @@ class AppCenterDelegateSwift: AppCenterDelegate {
     return "Internal"
   }
 
+  func configURL() -> String {
```
Comment (guperrot) [inline Sasquatch/SasquatchSwift/AppCenterDelegateSwift.swift]: Remove that

File: AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.m [hunk 4]
```diff
@@ -64,6 +64,7 @@ - (id)initWithBaseUrl:(NSString *)baseUrl
     _callsRetryIntervals = retryIntervals;
     _apiPath = apiPath;
     _maxNumberOfConnections = maxNumberOfConnections;
+    _baseURL = baseUrl;
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Ingestion/MSHttpIngestion.m]: I know you had a discussion about `URL` vs `Url` but please use `Url` for now.

File: AppCenter/AppCenter/Model/MSDevice.m [hunk 5]
```diff
@@ -1,5 +1,5 @@
-#import "MSACModelConstants.h"
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Model/MSDevice.m]: This isn't related change but format. I'd prefer revert this change because it has conflicts with `develop` branch.

File: AppCenter/AppCenter/Model/MSWrapperSdk.m [hunk 6]
```diff
@@ -1,5 +1,5 @@
-#import "MSACModelConstants.h"
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Model/MSWrapperSdk.m]: This isn't related change but format. I'd prefer revert this change because it has conflicts with `develop` branch.

File: AppCenterIdentity/AppCenterIdentity/Internals/Ingestion/MSIdentityConfigIngestion.h [hunk 7]
```diff
@@ -15,12 +15,12 @@ NS_ASSUME_NONNULL_BEGIN
...
  * Initialize the Ingestion.
  *
- * @param baseUrl Base url.
+ * @param baseURL Base URL.
  * @param appSecret A unique and secret key used to identify the application.
  *
  * @return An ingestion instance.
  */
-- (id)initWithBaseUrl:(nullable NSString *)baseUrl appSecret:(NSString *)appSecret;
+- (id)initWithBaseURL:(nullable NSString *)baseURL appSecret:(NSString *)appSecret;
```
Comment (jaelim-ms) [inline AppCenterIdentity/AppCenterIdentity/Internals/Ingestion/MSIdentityConfigIngestion.h]: Use `Url`.

File: AppCenterIdentity/AppCenterIdentity/Internals/MSIdentityPrivate.h [hunk 8]
```diff
@@ -32,6 +23,11 @@ static NSString *const kMSIdentityMSALAccountHomeAccountKey = @"MSIdentityMSALAc
  */
 @property(nonatomic, nullable) MSIdentityConfig *identityConfig;
 
+/**
+ * Base URL of the remote configuration file.
+ */
+@property(atomic, copy, nullable) NSString *configURL;
```
Comment (jaelim-ms) [inline AppCenterIdentity/AppCenterIdentity/Internals/MSIdentityPrivate.h]: `Url` please.

File: AppCenterIdentity/AppCenterIdentity/MSIdentity.h [hunk 9]
```diff
@@ -39,13 +39,13 @@ static NSString *const MSIdentityErrorDescriptionKey = @"MSIdentityErrorDescript
...
  *
- * @param url  The url with parameters.
+ * @param URL  The URL with parameters.
  *
  * @return `YES` if the URL is intended for App Center Identity and your application, `NO` otherwise.
  *
  * @discussion Place this method call into your app delegate's openURL method.
  */
-+ (BOOL)openURL:(NSURL *)url;
++ (BOOL)openURL:(NSURL *)URL;
```
Comment (jaelim-ms) [inline AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: `url`. I know it isn't consistent but let's do not change casing here.

File: AppCenterIdentity/AppCenterIdentity/MSIdentity.h [hunk 10]
```diff
@@ -59,6 +59,13 @@ static NSString *const MSIdentityErrorDescriptionKey = @"MSIdentityErrorDescript
  */
 + (void)signOut;
 
+/**
+ * Sets the base URL for the remote configuration file.
+ *
+ * @param configURL The base URL of the remote configuration file.
+ */
++ (void)setConfigURL:(NSString *)configURL;
```
Comment (jaelim-ms) [inline AppCenterIdentity/AppCenterIdentity/MSIdentity.h]: `Url`

File: Sasquatch/Sasquatch/Constants.h [hunk 11]
```diff
@@ -36,3 +36,5 @@ static int const kMSStoragePageSize = SQLITE_DEFAULT_PAGE_SIZE;
 #else
 static int const kMSStoragePageSize = 4096;
 #endif
+
+static const NSString *kMSIntConfigUrl = @"https://config-integration.dev.avalanch.es";
```
Comment (jaelim-ms) [inline Sasquatch/Sasquatch/Constants.h]: `kMSIntegrationConfigUrl`?
